pkg/apiserver/interfaces/api/pipeline.go

```diff
@@ -49,168 +43,211 @@ const (
+	ws.Route(ws.GET("/{projectName}/pipelines/{pipelineName}/runs/{runName}/log").To(n.getPipelineRunLog).
 		Doc("get pipeline run log").
-		Param(ws.QueryParameter("step", "query by specific id").DataType("string")).
+		Param(ws.QueryParameter("step", "query by specific step name").DataType("string")).
 		Returns(200, "OK", apis.GetPipelineRunLogResponse{}).
 		Returns(400, "Bad Request", bcode.Bcode{}).
+		Filter(n.RBACService.CheckPerm("project/pipeline/pipelineRun", "detail")).
 		Writes(apis.GetPipelineRunLogResponse{}).Do(meta, projParam, pipelineParam, runParam))
 
 	// get pipeline run output
-	ws.Route(ws.GET("/{pipelineName}/runs/{runName}/output").To(p.getPipelineRunOutput).
+	ws.Route(ws.GET("/{projectName}/pipelines/{pipelineName}/runs/{runName}/output").To(n.getPipelineRunOutput).
 		Doc("get pipeline run output").
-		Param(ws.QueryParameter("step", "query by specific id").DataType("string")).
+		Param(ws.QueryParameter("step", "query by specific step name").DataType("string").Required(true)).
 		Returns(200, "OK", apis.GetPipelineRunOutputResponse{}).
 		Returns(400, "Bad Request", bcode.Bcode{}).
+		Filter(n.RBACService.CheckPerm("project/pipeline/pipelineRun", "detail")).
 		Writes(apis.GetPipelineRunOutputResponse{}).Do(meta, projParam, pipelineParam, runParam))
 
+	// get pipeline run input
+	ws.Route(ws.GET("/{projectName}/pipelines/{pipelineName}/runs/{runName}/input").To(n.getPipelineRunInput).
+		Doc("get pipeline run input").
+		Param(ws.QueryParameter("step", "query by specific step name").DataType("string").Required(true)).
+		Returns(200, "OK", apis.GetPipelineRunInputResponse{}).
+		Returns(400, "Bad Request", bcode.Bcode{}).
+		Filter(n.RBACService.CheckPerm("project/pipeline/pipelineRun", "detail")).
+		Writes(apis.GetPipelineRunInputResponse{}).Do(meta, projParam, pipelineParam, runParam))
+
+	ws.Filter(authCheckFilter)
+}
+
+// GetWebServiceRoute is the implementation of pipeline Interface
+func (n *pipelineAPIInterface) GetWebServiceRoute() *restful.WebService {
+	tags := []string{"pipeline"}
+	meta := func(builder *restful.RouteBuilder) {
+		builder.Metadata(restfulspec.KeyOpenAPITags, tags)
+	}
+
+	ws := new(restful.WebService)
+
+	ws.Path(versionPrefix+"/pipelines").
+		Consumes(restful.MIME_XML, restful.MIME_JSON).
+		Produces(restful.MIME_JSON, restful.MIME_XML).
+		Doc("api for project manage")
+
+	ws.Route(ws.GET("").To(n.listPipelines).
+		Doc("list pipelines").
+		Param(ws.QueryParameter("query", "Fuzzy search based on name or description").DataType("string")).
+		Param(ws.QueryParameter("projectName", "query pipelines within a project").DataType("string")).
+		Param(ws.QueryParameter("detailed", "query pipelines with detail").DataType("bool").DefaultValue("true")).
+		Returns(200, "OK", apis.ListPipelineResponse{}).
+		Returns(400, "Bad Request", bcode.Bcode{}).
+		Writes(apis.ListPipelineResponse{}).Do(meta))
+
 	ws.Filter(authCheckFilter)
 	return ws
 }
 
+type pipelineAPIInterface struct {
```

Comment:
```suggestion
type pipelineAPI struct {
```

This is an struct not interface.

Reply:
This `Interface` seems to mean "network interface" not "interface in golang". And this is the naming rules for all other similar struct.

Reply:
This naming rule seems confusion, it's bad.

Reply:
We can change the naming uniformly. How do you like that @barnettZQG